### DISCUSSION TITLE
DIA-NN (v1.9) search with in-silico library discrepancy

### DISCUSSION THREAD

**Discussion by Cambridge27** [vdemichev/DiaNN#1272] [Q&A]
Hi,
i have generated in-silico library using FASTA file having Unimod:43 and Unimod:7 modification to map glycosylation sites (DeGly). The settings for library are att. The samples were run as nonDeGly and DeGly and analysed in DIA-NN using the generated library or separate analysis was done, where non-DeGly samples were run without library and only DeGly samples were run with library. The output results are very different. The protein ID's  generated using library for  non-DeGly samples  was 1800 and when used without library was 6000. The samples undergone enzymatic DeGly resulted in 1000 ID's when all samples together were run (no MBR) and 500 when only DeGly samples were run, all using library. The similar thing happened when i did phospho library for phospho search, two independent analysis gave two very different results. So, would you be able to advise please as i might have wrong settings

**Comment by vdemichev** [comment]
Hi,

Can you please share all the logs?

Best,
Vadim

**Comment by Cambridge27** [comment]
Hi,
The logs are attached. 4 samples were deglycosylated  (DeGly) and 4 native (glycosylated, Gly)). I have run  them together (using library) or separate, such as Gly without library  and DeGly with libray. Hope it is clear.  Am i creating library wrong?
[https://res.cdn.office.net/assets/mail/file-icon/png/txt_16x16.png]DeGly.log.txt<https://charlesriverlabs-my.sharepoint.com/:t:/p/cr234012/EfvkfGR3xBBPmYS1WKUtuAgB1PAeDGI-y3BQv0K3eegjkg>
[https://res.cdn.office.net/assets/mail/file-icon/png/txt_16x16.png]DeGly_and_Gly.log.txt<https://charlesriverlabs-my.sharepoint.com/:t:/p/cr234012/EcCPPPdJPTJFmQi43DTVVGsB_zUkuF223wfR7ONC--ZOKQ>
[https://res.cdn.office.net/assets/mail/file-icon/png/txt_16x16.png]Gly.log.txt<https://charlesriverlabs-my.sharepoint.com/:t:/p/cr234012/EXc7RBZb6_NFsVYxk5xRYMMBesdo105GQL7g7jet_7qTVg>

Many thanks for your help, kind regards, Ruza
________________________________

**Comment by vdemichev** [comment]
Hi Ruza,

Sorry, the above links do not seem to work. Can you please just attach the files to the github post?

Best,
Vadim

**Comment by Cambridge27** [comment]
[DeGly.log.txt](https://github.com/user-attachments/files/17872106/DeGly.log.txt)
[DeGly_and_Gly.log.txt](https://github.com/user-attachments/files/17872107/DeGly_and_Gly.log.txt)
[Gly.log.txt](https://github.com/user-attachments/files/17872108/Gly.log.txt)

**Reply by Cambridge27** [reply]
Sorry, these are the logs.
Thanks so much for the help

kind regards, Ruza

**Comment by vdemichev** [comment]
Need to address

> WARNING: it is strongly recommended to first generate an in silico-predicted library in a separate pipeline step and then use it to process the raw data, now without activating FASTA digest

Please also use MBR.

Why are you specifying glycosilation as a GG adduct on Lysine (UniMod:121)?

> The samples undergone enzymatic DeGly resulted in 1000 ID's when all samples together were run (no MBR) and 500 when only DeGly samples were run, all using library

Libraries are different, e.g. enabmed M(ox) in some but not other runs.

**Comment by Cambridge27** [comment]
Many thanks. Can I please clarify something with you.
For example, I am interested in phosphor proteomics.

To generate library  I would use the settings below? For phospho library generation do I check other modifications or I check them in data search when I generate library only by ticking phospho

***@***.***


And then for data search, I only include: raw files, library, FASTA file (that was used to generate library) and check all modifications that were used to generate librarye?
***@***.***

OR do I generate library only with phosphor modification and then in data search include all usual?



Kind regards, Ruza

**Comment by vdemichev** [comment]
> To generate library  I would use the settings below?

The images failed to paste, but please just follow recommendations here https://github.com/vdemichev/DiaNN?tab=readme-ov-file#ptms-and-peptidoforms.

> And then for data search, I only include: raw files, library, FASTA file (that was used to generate library) and check all modifications that were used to generate librarye?

During search, specifying variable modifications means you want DIA-NN to localise them (and this affects a number of other things). So if you only care about localisation of phosho, only specify phospho.

**Reply by Cambridge27** [reply]
Hi,
i am coming back in regards to phosphoproteomics.
i have created library (1missCleav.log) and then set the search on two replicates of same sample (250101.log).
in the report file (250101), not sure how those result became the file with 99% probability (250101.phosphosites_99 file). with only one phospho peptide reported, While "250101" file state 0.99 or 1 probability for majority peptides. We wera running these in as TMT as well and analyzed by PD and got cca 7000 phosphopeptides. So i am setting something wrong and not sure what. 
In addition, would you kindly advise which column to use to filter data from "250101" report? 
Is there any explanation on these:
PTM.Informative | PTM.Specific | PTM.Localising | PTM.Q.Value | PTM.Site.Confidence | Lib.PTM.Site.Confidence
PEP | Global.Q.Value

*Just a reminder i know what is Q value and what is PEP, just not sure what to look at in the report

many thanks in advance, Ruza
[1missCleav.log.txt](https://github.com/user-attachments/files/18289510/1missCleav.log.txt)
[250101.log.txt](https://github.com/user-attachments/files/18289511/250101.log.txt)
[250101.xlsx](https://github.com/user-attachments/files/18289512/250101.xlsx)

**Reply by vdemichev** [reply]
Hi Ruza,

Can you please attach (or share in some other way) the 90% matrix and the .parquet report?

Best,
Vadim

**Reply by Cambridge27** [reply]
Hi,
here is 90% file.
[250101.phosphosites_90.xlsx](https://github.com/user-attachments/files/18289694/250101.phosphosites_90.xlsx)
Not sure what is happening with PARQUET, i can not upload it as it is not supported? when i try to save it as text it turns into symbols?
many thanks, Ruza

**Reply by vdemichev** [reply]
Why .xlsx and not .tsv? 

> i can not upload it as it is not supported

Can upload to any file sharing service, e.g. your Google Drive, OneDrive, etc.

**Reply by vdemichev** [reply]
About the log, it shows 'deep learning' option checked while you are loading a predicted library. That is in the beginning DIA-NN spends time to replace data in the library with the same data freshly predicted, i.e. normally this should not be done. Don't think it affects the output here though.

**Reply by Cambridge27** [reply]
Hi,
i have tried again and it is not working. Would you kindly advise for other option to share parquet file. Maybe it is my company filter. Would you maybe have an email that can be used for this purposes?
kind regards, Ruza

**Reply by vdemichev** [reply]
> Would you maybe have an email 

https://github.com/vdemichev/DiaNN?tab=readme-ov-file#support

**Reply by Cambridge27** [reply]
Hi, i have sent the requested file via email vadim.demichev@gmail.com. hope it will work.
i am really keen to have phosphoproteomics work  on DIA-NN.
many thanks

**Reply by vdemichev** [reply]
Hi Ruza,

Many thanks for sharing (and helping to improve DIA-NN! :)). I was able to reproduce the problem, it occurs due to reannotation being enabled. We will investigate and fix this, but workaround is very simple in this case (since reannotation is not really needed):
- For the analysis step, enable MBR and make sure 'Reannotate' and 'Deep learning' checkboxes are not checked. 

In general (not related to this issue), if you use DIA-NN 1.9.2, then RAM usage will be much lower and most likely will be able to enable up to 3 var mods, which will yield more phosphopeptides.

Best,
Vadim

**Reply by Cambridge27** [reply]
Many thanks for the reply. i will re-analyse with "Reannotate" and "Deep learning" unchecked. 
I am puzzled by one more thing.
When i create library, i create with only --var-mod UniMod:21,79.966331,STY in Additional Options box. That right?
But when i do sample analysis i keep --var-mod UniMod:21,79.966331,STY in Additional Options box, plus checking modifications from Precursor Ion Generation box in GUI? Is that correct?
I would need to check at least Cystein?
What would happen if i would generate library with Unimod21 and Cystein carbamidomethylation? would DIA-NN crash?

Also when analysing samples, is there any benefit in checking in the Output window "Generate spectral Library" or "generate PROSIT input from FASTA or sp. lib"? I assume they take some RAM, but not necessary for sample analysis or am i wrong?

many thanks

**Reply by Cambridge27** [reply]
H, i have analyzed the data as suggested and also included four raw files for analysis (when i put all 12 raw files, DIA-NN crashes).
I am not sure how one gets from report to phosphosites_90. It is not based on PTM.Site.Confidence from report file. Am i missing something?
I will send the files via email. i have blockage again.

many thanks, Ruza

**Reply by vdemichev** [reply]
Hi Ruza,

> When i create library, i create with only --var-mod UniMod:21,79.966331,STY in Additional Options box

You don't need to use 'Additional options' at all for phospho, the 'Phospho' checkbox is sufficient.

> --var-mod UniMod:21,79.966331,STY in Additional Options box, plus checking modifications from Precursor Ion Generation box in GUI

These have exactly same effect, i.e. having both the --var-mod in additional options and the checkbox has the same effect as just having one.

> What would happen if i would generate library with Unimod21 and Cystein carbamidomethylation? would DIA-NN crash?

Why would it? C(cam) is also on by default.

>  is there any benefit in checking in the Output window "Generate spectral Library" or "generate PROSIT input from FASTA or sp. lib"?

'Generate sectral library' will get checked automatically if you check MBR (which should be on). Generate Prosit input if you need input for the tool Prosit, other than that it's not doing anything.

> when i put all 12 raw files, DIA-NN crashes

Can you please share the log? Or just runs out of RAM (will be a lot better in 1.9.2 and better still in 2.0).

> I am not sure how one gets from report to phosphosites_90

But if you disabled reannotation, now the phosphosites_90 table should be correctly generated by DIA-NN?
Otherwise, you would need to write a custom script based on site-specific localisation confidence scores reported by DIA-NN in the main report. But the whole purpose of phosphosite matrices is for DIA-NN to do this automatically :) 

> It is not based on PTM.Site.Confidence from report file.

Relevant columns are Site.Occupancy.Probabilities and Protein.Sites, in case you don't want to use phosphosites_90 matrix but want to generate it youself.

Best,
Vadim

**Reply by Cambridge27** [reply]
Hi, i apologize again. One more question on generating library. 
i want to check, phospho, oxidation (m), deamidation as var.mod  and cysteine CAM as fixed. Phospho is my main goal.
When i generate library, do I:
1. generate library with only phospho set up as var. mod? and afterward during analysis check other modifications of interest?
2. use all modifications of interest to generate library? 

i was not very clear before with this question.

kind regards, Ruza
